source/crc32.c

```diff
@@ -0,0 +1,73 @@
+#include <aws/cal/crc32.h>
+#include <aws/cal/hash.h>
+#include <aws/checksums/crc.h>
+
+/* hash is object, vtable is methods, hash properties (not vtable) are data?*/
+
+static void s_destroy(struct aws_hash *hash);
+static int s_update(struct aws_hash *hash, const struct aws_byte_cursor *to_hash);
+static int s_finalize(struct aws_hash *hash, struct aws_byte_buf *output);
+/* why do we need to define and implement the functions separatly? */
+/* can we reuse some of these as defaults defined in hash.c?*/
+
+/* does this have to be defined globaly? I'd like to use a different update function depending on a parameter */
+static struct aws_hash_vtable s_vtable = {
```

Comment:
go ahead and make this s_crc32_vtable, we know that you're going to add s_crc32c_vtable next, and they share a lot.